src/IO/ITKImageInputFileFormat.cxx

```diff
@@ -278,40 +303,21 @@ construct_empty_stir_image(typename ITKImageType::Pointer itk_image,
    This method expects that itk_image is already oriented to be consistent with
    STIR x, y, z axes.
 */
-void copy_ITK_data_to_STIR_image(const ITKImageSingle::Pointer itk_image,
-                                 STIRImageSingle* stir_image_ptr)
+template<typename ITKImageType, typename STIRImageType>
+void copy_ITK_data_to_STIR_image(const typename ITKImageType::Pointer itk_image,
+                                 STIRImageType* stir_image_ptr,
```

Comment:
why not a const ref as opposed to a pointer?

Reply:
I've made this a reference, but it can't be const as we update the contents.